_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+* If your API Gateway is going to be accessed from the same AWS region, the Regional API Gateway will have less latency.
+
+# Integration with Private VPCs
+A big downside to API Gateway has alweays been that any HTTP backend behind the API Gateway needed to be publicly accessible. And because the public IP address of the API Gateway is unknown or unpredictable, IP whitelisting at the backend system was not a viable option. To solve this issue, Amazon [introduced](https://aws.amazon.com/about-aws/whats-new/2017/11/amazon-api-gateway-supports-endpoint-integrations-with-private-vpcs/) integration with private VPCs.
+
+The connection between your API Gateway and your private resources needs two parts:
+* A private Network Load Balancer (NLB) in your VPC in front of the resources you want to access
+* A API Gateway VPC Link that points to that NLB
+
+![VPC Link](/assets/posts/2018-09-13-API-Gateway-Types-Compared/vpc-link.png)
+
+Hurray! With this feature, API Gateway no longer demands your resources to be public and the security of your VPC-based resources is greatly enhanced.
+
+# Private API Gateways
+The integration with private VPC endpoints is an awesome improvement. However, there is still one major issue regarding the security of API Gateway: it is always publicly available. Of course you can implement authentication, rate limiting, API keys, WAF and [resource policies](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-resource-policies.html), but this still feels like a hack when your API Gateway is only used by your VPC-based resources or over AWS VPN. To solve this issue, AWS [introduced](https://aws.amazon.com/blogs/compute/introducing-amazon-api-gateway-private-endpoints/) the **Private** API Gateway in June 2018.
+
+The Private API Gateway is not publicly accessible or resolvable. Instead, it can only be accessed from within your VPC. To achieve this, AWS uses the [PrivateLink](https://aws.amazon.com/about-aws/whats-new/2017/11/introducing-aws-privatelink-for-aws-services/) technology. Diagram time!
+
+![Private API](/assets/posts/2018-09-13-API-Gateway-Types-Compared/private-api.png)
+
+AWS PrivateLink is an incredible addition to the AWS technology stack. It allows you to access many AWS Services (e.g. S3, KMS, SNS, CodeBuild and API Gateway) over a network interface in your own VPC. Take S3 as an example: before PrivateLink, any traffic from an EC2 instance to an S3 bucket would travel over the public internet. With PrivateLink, all traffic to S3 is routed to the Elastic Network Adapter(s) you've created in your own VPC. From this interface, the traffic will only traverse Amazon's internal network to S3. And as a bonus, the PrivateLink endpoint for S3 allows you to specify a policy which buckets can be accessed. This is a *great* security feature, because this effectively solves the problem of unauthorized data extraction to unknown buckets.
```

Comment:
`Take S3 as an example:`
Not so great example as there use to be VPC Endpoints for S3 already which basically provided the same result

Reply:
Well, it is a very clear example of why you would want to use PrivateLink.. and I don't think it matters that it already existed before. The point I'm trying to make is still the same.

Reply:
But in the context on why PrivateLink is superduper awesome S3 is the least interesting as it already had such feature